File: StackUsingLinkedList.java
package l13;
import l12.Nodes;

public class StackUsingLinkedList<T> {
	
	private Nodes<T> node;
	private int length;
	
	public  StackUsingLinkedList() {
		length=0;
		node=null;
		
	}
	
	public int size(){
		return length;
	}
	
	public void print(){
		Nodes<T> temp;
		temp=node;
		
		while(temp!=null){
			System.out.print(temp.data+"-->");
			temp=temp.next;
		}
	}
	
	public boolean isEmpty(){
			
		if(length==0){
				return true;
		}
		return false;
	}
	
	public void push(T element){
		Nodes<T> temp=new Nodes<T>(element);
	
		if(this.size()==0){
			temp.next=null;
			node=temp;
		}else{
			temp.next=node;
			node=temp;
		}
		length++;
}
	public T pop()throws StackEmptyException{
		if(node==null){
			StackEmptyException e = new StackEmptyException("pop");
			throw e;
		}
		T elementToReturned;
		elementToReturned= node.data;
		node=node.next;
		length--;
		return elementToReturned;
	}
	
	public T top()throws StackEmptyException{
		if(node==null){
			StackEmptyException e=new StackEmptyException("Top");
			throw e;
		}
		return node.data;
	}
}

File: StackUsingLinkedList.cpp
#include <iostream>

using namespace std;

struct node{
node *next;
int data;
};

class myStack{
    node *top;

    public:
    myStack(){
        top =  NULL;
    }

    void push(int value){
        node* temp = new node;
        temp -> data = value;

        if(top == NULL){
            top = temp;
            top -> next = NULL;
        }else{
            temp -> next = top;
            top = temp;
        }
        cout<<"Value Pushed \n";
    }

    void pop(){
        node *temp = new node;
        if(top == NULL){
            cout<<"Already Empty Stack "<<endl;
        }else{
            temp = top;
            top = top -> next;
            cout<<top -> data<<"popped out of Stack \n";
        }

    }

    void display(){
        node *temp = top;

        while(temp != NULL){
            cout<<temp -> data<<"\t";
            temp = temp -> next;
        }
        cout<<endl;
    }
};
int main()
{
    myStack s1;
    s1.push(12);
    s1.display();
    s1.push(13);
    s1.display();
    s1.push(14);
    s1.display();
    s1.pop();
    s1.display();
}
